### PR TITLE
[cicd] Implementing pytest-split for databricks test

### PR DESCRIPTION
**Pull request by bushwhackr**
## Description

+ databricks-unity test is split into 3 runners (~1hr 15min to 45min)
+ fixed getting spark logs for spark integration test
+ fixed overriding test file issues due to matrix
+ overriding changelog entries and dependency changes github comment workflow

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: .github/workflows/test_gen.yaml [hunk 1]
```diff
@@ -319,13 +320,18 @@ jobs:
...
       matrix:
         python-version:
-        - 3.8.12
+          - 3.8.12
+        pytest-group:
+          - 1
+          - 2
+          - 3
+    env:
+      PYTEST_SPLITS: 3  # This needs to be EQ the count of strategy.matrix.pytest-group
```
Comment (bushwhackr) [inline .github/workflows/test_gen.yaml:334]: Only splitting into 3 runners for databricks unity tests

**Review by kchua78 — APPROVED** [review]

File: .github/workflows/test_gen.yaml [hunk 2]
```diff
@@ -223,30 +224,29 @@ jobs:
...
+        path: spark-thrift-${{ matrix.spark-version }}.log
         retention-days: 5
-    - name: Check test status
-      run: |
-        EXIT_CODE=$(cat status.txt)
-        if [ $EXIT_CODE -ne 0 ]; then
-          exit $EXIT_CODE
-        fi
+
+    # NOTE: Only holding 1 test result to be used in pytest report
```
Comment (YS-L) [inline .github/workflows/test_gen.yaml:249]: Curious what does this mean? Are the spark changes required?

File: .github/workflows/test_gen.yaml [hunk 3]
```diff
@@ -454,10 +469,18 @@ jobs:
...
+        name: python-${{ matrix.python-version }}-pytest.xml.4.1
+    - name: Download integration test results (databricks-unity_2)
+      uses: actions/download-artifact@v4
+      with:
+        name: python-${{ matrix.python-version }}-pytest.xml.4.2
+    - name: Download integration test results (databricks-unity_3)
       uses: actions/download-artifact@v4
       with:
-        name: python-${{ matrix.python-version }}-pytest.xml.4
+        name: python-${{ matrix.python-version }}-pytest.xml.4.3
```
Comment (YS-L) [inline .github/workflows/test_gen.yaml:483]: I wonder if it's possible to glob multiple files with a pattern so we don't need to repeat this for every split (and actually even for every platforms)?

**Comment by YS-L** [comment]
This looks great. Looks like the coverage report is showing up again now. Is it possible to get the coverage reports from a recent previous PR to compare there's no unexpected changes?

File: .github/workflows/test_gen.yaml (hunk 2 above)
Comment (bushwhackr) [reply .github/workflows/test_gen.yaml:249]: we are testing on 2 different spark version. We only save the last one that finished, as we override the previous artifact that is generated. That pytest.xml and coverage is the only one being used in the merging step.

File: .github/workflows/test_gen.yaml (hunk 3 above)
Comment (bushwhackr) [reply .github/workflows/test_gen.yaml]: Testing it out now

**Comment by bushwhackr** [comment]
> This looks great. Looks like the coverage report is showing up again now. Is it possible to get the coverage reports from a recent previous PR to compare there's no unexpected changes?

I have another PR to fix the coverage report issue. It was due to the validate step being skipped due to databricks test being disabled.
I'll run the test in a separate PR with and without pytest-split and see if there are any issues.

**Pull request merged**